crates/bundler/src/bundler.rs

```diff
@@ -78,4 +101,142 @@ impl Bundler {
+
+        let _bundle_signer = env::var("FLASHBOTS_IDENTIFIER").expect("FLASHBOTS_IDENTIFIER environment variable is not set");
+
+        let bundle_signer = _bundle_signer.parse::<LocalWallet>()?;
+
+        let mut fb_middleware = FlashbotsMiddleware::new(
+                    provider.clone(),
+                    Url::parse(relay_endpoint.clone())?,
+                    bundle_signer.clone(),
+                );
+        fb_middleware.set_simulation_relay(Url::parse(relay_endpoint.clone()).unwrap());
+
+        let client = Arc::new(
+            SignerMiddleware::new(
+                fb_middleware,
+                self.wallet.signer.clone(),
+            )
+        );
+
+        let ep = EntryPointAPI::new(self.entry_point, client.clone());
+
+        let nonce = client
+            .clone()
+            .get_transaction_count(self.wallet.signer.address(), None)
+            .await?;
+        let mut tx: TypedTransaction = ep
+            .handle_ops(
+                uos.clone().into_iter().map(Into::into).collect(),
+                self.beneficiary,
+            )
+            .tx
+            .clone();
+        tx.set_nonce(nonce).set_chain_id(self.chain.id());
+
+        trace!("Sending transaction to the execution client: {tx:?}");
+
+        // Sign the tx
+        let typed_tx = TypedTransaction::Eip1559(tx.clone().into());
+        let raw_signed_tx = match client.signer().sign_transaction(&typed_tx).await {
+            Ok(tx) => typed_tx.rlp_signed(&tx),
+            Err(e) => {
+                return Err(anyhow::anyhow!(
+                    "Failed to sign transaction: {:?}",
+                    e
+                ))
+            }
+        };
+
+        // Add tx to Flashbots bundle
+        let mut bundle_req = BundleRequest::new();
+        bundle_req = bundle_req.push_transaction(raw_signed_tx);
+
+        // Simulate the Flashbots bundle
+        let block_num = client.get_block_number().await?;
+        bundle_req = bundle_req
+            .set_block(block_num + 1)
+            .set_simulation_block(block_num)
+            .set_simulation_timestamp(0);
+        let simulated_bundle = client.inner().simulate_bundle(&bundle_req).await?;
+
+        // Currently there's only 1 tx per bundle 
+        for tx in simulated_bundle.transactions {
+            trace!("Simulate bundle: {:?}", tx);
+            if let Some(err) = &tx.error { 
+                return Err(anyhow::anyhow!(
+                    "Transaction failed simulation with error: {:?}",
+                    err
+                ));
+            }
+            if let Some(revert) = &tx.revert { 
+                return Err(anyhow::anyhow!(
+                    "Transaction failed simulation with revert: {:?}",
+                    revert
+                ));
+            }
+        };
+
+        // Send the Flashbots bundle and check for status
+        let handle: JoinHandle<Result<(bool, H256), anyhow::Error>> = tokio::spawn(
```

Comment:
What is the reason to spawn a tokio thread here?

Reply:
The requirements I had was this
-the bundler should be able to send the bundle to relays while processing new UserOperations from the alt mempool. To not block the main thread processing the latest `UserOperations`, we need a thread that handles sending and monitoring the status bundles that have been sent
-the bundler should be able to send the same bundle to multiple relays in the future to maximize the chances of inclusion. Sending and monitoring status to multiple relays is a quite complex operation, hence should not block the main thread. When we start supporting multiple relays, we can change this part of the code to something like this
```
let handles: Vec<_> = RELAY_ENDPOINTS
        .len()
        .map(|i| {
            tokio::spawn(async move {
                println!("Starting task {}", i);
                // sending bundles
                println!("Completed task {}", i);
            })
        })
        .collect();

    for handle in handles {
        let _ = handle.await;
    }
```